anydex/test/wallet/monero/test_xmr_wallet.py

```diff
@@ -0,0 +1,535 @@
+        mock_wallet.balance = lambda unlocked: 20.2
+
+        mock_transaction = MockObject()
+        mock_transaction.hash = TEST_HASH
+
+        mock_wallet.transfer = lambda *_, **__: mock_transaction
+
+        w.wallet = mock_wallet
+
+        result = await w.transfer(20.1, 'test_address',
+                                  payment_id='test_id',
+                                  priority=1,
+                                  unlock_time=0)
+        self.assertEquals(TEST_HASH, result.result())
+
+    async def test_transfer_wallet_not_enough(self):
+        """
+        Attempt XMR transfer in case not enough XMR available.
+        """
+        w = MoneroWallet()
+        succeed_backend()
+        await w.create_wallet()
+
+        mock_wallet = MockObject()
+        mock_wallet.refresh = lambda *_: None
+        mock_wallet.balance = lambda unlocked: 20.2
+
+        w.wallet = mock_wallet
+
+        self.assertAsyncRaises(InsufficientFunds, w.transfer(47.8, 'test_address',
+                                                             payment_id='test_id',
+                                                             priority=1,
+                                                             unlock_time=0))
+
+    async def test_transfer_multiple_no_wallet(self):
+        """
+        Make multiple transfers, but no wallet initialized.
+        """
+        w = MoneroWallet()
+
+        transfers = [
+            (TEST_ADDRESS, Decimal('20.2')),
+            (TEST_ADDRESS, Decimal('7.8'))
+        ]
+
+        self.assertAsyncRaises(WalletConnectionError, w.transfer_multiple(transfers, priority=3))
+
+    async def test_transfer_multiple_wallet(self):
+        """
+        Make multiple transfers at once, enough balance.
+        """
+        w = MoneroWallet()
+
+        transfers = [
+            (TEST_ADDRESS, Decimal('20.2')),
+            (TEST_ADDRESS, Decimal('7.8'))
+        ]
+
+        mock_wallet = MockObject()
+        mock_wallet.refresh = lambda: None
+        mock_wallet.balance = lambda **_: 57.3
+        mock_wallet.transfer_multiple = \
+            lambda *_, **__: [(Transaction(hash=TEST_HASH), Decimal('20.2')),
+                              (Transaction(hash=TEST_HASH), Decimal('7.8'))]
+
+        w.wallet = mock_wallet
+
+        hashes = await w.transfer_multiple(transfers, priority=3)
+        self.assertEquals(2, len(hashes.result()))
+
+    async def test_transfer_multiple_wallet_not_enough(self):
+        """
+        Make multiple transfers at once, but not enough balance.
+        """
+        w = MoneroWallet()
+
+        transfers = [
+            (TEST_ADDRESS, Decimal('20.2')),
+            (TEST_ADDRESS, Decimal('7.8'))
+        ]
+
+        mock_wallet = MockObject()
+        mock_wallet.refresh = lambda: None
+        mock_wallet.balance = lambda **_: 21.3
+        w.wallet = mock_wallet
+
+        self.assertAsyncRaises(InsufficientFunds, w.transfer_multiple(transfers, priority=3))

```

Comment:
I get an error : `RuntimeWarning: coroutine 'MoneroWallet.transfer' was never awaited`.
It might be because `self.assertAsyncRaises` is a coroutine and you do not await it.

Reply:
```suggestion
        self.assertAsyncRaises(InsufficientFunds, await w.transfer_multiple(transfers, priority=3))

```

Reply:
This then gives me this error 
```
Future exception was never retrieved
future: <Future finished exception=InsufficientFunds('Insufficient funds found in Monero wallet for all transfers')>
anydex.wallet.wallet.InsufficientFunds: Insufficient funds found in Monero wallet for all transfers

```

Reply:
But i think the test works now when awaited

Reply:
Checks are now passing. I am also not able  to reproduce the `Runtimewarning` issue, so I must assume it is now fixed. The warning @osheasha mentioned appears for every test for me. Even those that do not test coroutines, I am assuming the issue is therefore related to something else, maybe the testing setup?